### PR TITLE
Add Jakarta REST client module for HttpClient

### PR DESCRIPTION
**Pull request by arturobernalg**
This PR adds httpclient5-jakarta-rest-client, a small Java 11+ client-side module that creates implementations of jakarta.ws.rs-annotated interfaces with Apache HttpClient underneath.

The c/]qurrent implementation is intentionally minimal and focused on the client side only.

### CODE REVIEW

File: httpclient5-jakarta-rest-client/src/test/java/org/apache/hc/client5/http/examples/RestClientExample.java [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+                                            "bob@example.com")
+                            };
+                            response.setCode(200);
+                            response.setEntity(new StringEntity(mapper.writeValueAsString(users), ContentType.APPLICATION_JSON));
+                        } else if ("POST".equals(method)) {
+                            final String body = EntityUtils.toString(request.getEntity());
+                            final User u = mapper.readValue(body, User.class);
+                            u.id = 3;
+                            response.setCode(201);
+                            response.setEntity(new StringEntity(mapper.writeValueAsString(u), ContentType.APPLICATION_JSON));
```
[the remark this answers is not in the document]
Comment (arturobernalg) [reply httpclient5-jakarta-rest-client/src/test/java/org/apache/hc/client5/http/examples/RestClientExample.java]: mmmmm

**Comment by ok2c** [comment]
@arturobernalg Was this change-set AI generated? I have an impression it was authored by someone who has no familiarity with our code base.

This is a start and a good proof of concept. Would you actually be willing to take your time and do it better?

**Comment by arturobernalg** [comment]
> Was this change-set AI generated? I have an impression it was authored by someone who has no familiarity with our code base.
> 
> This is a start and a good proof of concept. Would you actually be willing to take your time and do it better?

@ok2c  Yes. I can rework it. If you have a specific direction in mind, please let me know

**Comment by arturobernalg** [comment]
@ok2c 

I reworked the patch to keep it small and direct, dropped the extra framework-like pieces, and aligned it more closely with the existing HttpClient style.
Is this closer to what you had in mind when you said it needed to be done better, or do you still see parts that feel off in terms of scope or design?

**Comment by ok2c** [comment]
@arturobernalg 

1. Please use the async client because we want the REST client to support HTTP/2 as well.  Please use it  _without_ resorting to content buffering in memory (do not use 'simple' message objects). 'Simple' APIs are not intended for production use, so we should not be using them internally. Use the new Jackson2 bindings from core.  
2. Please do not re-implement the percent codec. What stops you from using what we already have? Not good enough? Improve it.
3. Please use `URIBuilder` to construct request URIs.

The whole thing still looks like it just got vibe-coded with some AI agent. There is no need to rush it. Take your time to do it right. I think this can be the most important contribution to the project since @rschmitt reactive module. So, _please_ take your time

**Comment by arturobernalg** [comment]
I haven't forgot of the PR.  I just waiting for this change [54135f6](https://github.com/apache/httpcomponents-core/commit/54135f6cacfeb28208c58b4dadd355f6b5a2dfad)

**Comment by ok2c** [comment]
> I haven't forgot of the PR. I just waiting for this change [54135f6](https://github.com/apache/httpcomponents-core/commit/54135f6cacfeb28208c58b4dadd355f6b5a2dfad)

@arturobernalg I do not understand how this can be holding you back. This change in core allows for a custom content-type for JSON content, which looks like a corner case to me. I do not see how it is relevant here.

**Comment by arturobernalg** [comment]
> > I haven't forgot of the PR. I just waiting for this change [54135f6](https://github.com/apache/httpcomponents-core/commit/54135f6cacfeb28208c58b4dadd355f6b5a2dfad)
> 
> @arturobernalg I do not understand how this can be holding you back. This change in core allows for a custom content-type for JSON content, which looks like a corner case to me. I do not see how it is relevant here.

@ok2c 
I removed the dependency on it in the Jakarta REST client and switched the POJO request body path to the current Jackson2 producer API. I also disabled the test that was asserting the custom JSON media type corner case for POJO bodies.

**Comment by ok2c** [comment]
> @ok2c I removed the dependency on it in the Jakarta REST client and switched the POJO request body path to the current Jackson2 producer API. I also disabled the test that was asserting the custom JSON media type corner case for POJO bodies.

@arturobernalg Great! I will do a thorough review either this weekend or the next one the latest. This is _the_ top priority change-set for me.

**Comment by ok2c** [comment]
@arturobernalg Could we use httpbin as a target for the example code to simplify it?

`curl -X GET "https://httpbin.org/get" -H  "accept: application/json"`

```
{
  "args": {},
  "headers": {
    "Accept": "application/json",
    "Accept-Encoding": "gzip, deflate, br, zstd",
    "Accept-Language": "en-US,en;q=0.5",
    "Host": "httpbin.org",
    "Priority": "u=0",
    "Referer": "https://httpbin.org/",
    "Sec-Fetch-Dest": "empty",
    "Sec-Fetch-Mode": "cors",
    "Sec-Fetch-Site": "same-origin",
    "User-Agent": "Mozilla/5.0 (X11; Linux x86_64; rv:140.0) Gecko/20100101 Firefox/140.0",
    "X-Amzn-Trace-Id": "Root=1-69edec00-494d54b55f97010c2add55f6"
  },
  "origin": "213.55.197.162",
  "url": "https://httpbin.org/get"
}
```

Looks good otherwise. I would like to try to re-work some of the internals and the exception handling but this can be done as a follow-up change.

**Comment by arturobernalg** [comment]
> @arturobernalg Could we use httpbin as a target for the example code to simplify it?
> 
> `curl -X GET "https://httpbin.org/get" -H "accept: application/json"`
> 
> ```
> {
>   "args": {},
>   "headers": {
>     "Accept": "application/json",
>     "Accept-Encoding": "gzip, deflate, br, zstd",
>     "Accept-Language": "en-US,en;q=0.5",
>     "Host": "httpbin.org",
>     "Priority": "u=0",
>     "Referer": "https://httpbin.org/",
>     "Sec-Fetch-Dest": "empty",
>     "Sec-Fetch-Mode": "cors",
>     "Sec-Fetch-Site": "same-origin",
>     "User-Agent": "Mozilla/5.0 (X11; Linux x86_64; rv:140.0) Gecko/20100101 Firefox/140.0",
>     "X-Amzn-Trace-Id": "Root=1-69edec00-494d54b55f97010c2add55f6"
>   },
>   "origin": "213.55.197.162",
>   "url": "https://httpbin.org/get"
> }
> ```
> 
> Looks good otherwise. I would like to try to re-work some of the internals and the exception handling but this can be done as a follow-up change.

@ok2c please check again

**Review by ok2c — APPROVED** [review]

**Comment by arturobernalg** [comment]
@ok2c I'll add the doc on the web module latter on.
